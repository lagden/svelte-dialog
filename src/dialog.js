/* istanbul ignore file */

import * as acts from './acts.js'
import Dialog from './Dialog.svelte'

export {
	Dialog,
	acts,
}
