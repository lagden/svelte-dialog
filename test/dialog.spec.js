/* globals describe, afterEach, test, expect */

import {cleanup, render} from '@testing-library/svelte'
import {Dialog, acts} from '../src/dialog.js'

describe('Dialog', () => {
	afterEach(cleanup)

	test('should match snapshot', done => {
		const {container} = render(Dialog, {
			props: {
				name: 'teste',
				title: 'Teste',
				message: 'Apenas um show.',
			},
		})
		acts.open('teste')

		setTimeout(() => {
			expect(container).toMatchSnapshot()
		}, 1000)

		setTimeout(() => {
			acts.close('teste')
			done()
		}, 1500)
	})
})
