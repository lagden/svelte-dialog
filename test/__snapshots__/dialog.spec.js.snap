// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should match snapshot 1`] = `
<body>
  <div>
    
  </div>
  <div
    class="_tadashi_svelte_dialog _tadashi_svelte_dialog__align_top svelte-1uuxjiv"
    style=""
  >
    <div
      class="_tadashi_svelte_dialog__box svelte-1uuxjiv"
    >
      <div
        class="_tadashi_svelte_dialog__content _tadashi_svelte_dialog__grid svelte-1uuxjiv"
      >
        <div
          class="_tadashi_svelte_dialog__header svelte-1uuxjiv"
        >
          <h3
            class="_tadashi_svelte_dialog__title svelte-1svxb87"
            slot="header"
          >
            Teste
          </h3>
        </div>
         
        <button
          class="_tadashi_svelte_dialog__close svelte-1uuxjiv"
          type="button"
        >
          Ã—
        </button>
         
        <div
          class="_tadashi_svelte_dialog__body svelte-1uuxjiv"
        >
          Apenas um show.
          
           
           
        </div>
      </div>
    </div>
  </div>
</body>
`;
